docs/encyclopedia/application-message-passing.mdx

```diff
@@ -342,16 +342,21 @@ The following content applies in every SDK except the Go SDK. See below.
 #### Exceptions in Signals
 
 In Signal handlers, throw [Application Failures](/references/failures#application-failure) only for unrecoverable errors, because the entire Workflow will fail.
+Similarly, allowing a failing Activity or Child Workflow to exhaust its retries, so that it throws an [Activity Failure](https://docs.temporal.io/references/failures#activity-failure) or [Child Workflow Failure](https://docs.temporal.io/references/failures#child-workflow-failure) will cause the entire Workflow to fail.
+Note that for Activities, this will only happen if you change the default Activity [Retry Policy](https://docs.temporal.io/encyclopedia/retry-policies), since by default they retry forever.
 If you throw any other exception, by default, it will cause a [Workflow Task Failure](/references/failures#workflow-task-failures). This means the Workflow will get stuck and will retry the handler periodically until the exception is fixed, for example by a code change.
 
 #### Exceptions in Updates
 
-When you want to fail the Update and for the client to receive that error:
+Doing any of the following will fail the Update and cause the client to receive the error:
```

Comment:
This may be missing:

* Bubble failure to send signal to child/external workflow
* Bubble any error/exception/panic configured as a workflow failure (configurable in Go, Java, Python, and .NET in different ways)

Reply:
I'm happy leaving those out here; the previous version wasn't exhaustive either. The main aim is to get readers to understand Temporal's high-level philosophy.